packages/devtools_app/lib/src/shared/common_widgets.dart

```diff
@@ -2259,3 +2260,78 @@ class ContextMenuButton extends StatelessWidget {
     );
   }
 }
+
+/// A Widget for displaying a setting field, that sets and integer value.
+class PositiveIntegerSetting extends StatefulWidget {
+  const PositiveIntegerSetting({
+    super.key,
+    required this.title,
+    required this.subTitle,
+    required this.notifier,
+  });
+
+  final String title;
+  final String subTitle;
+  final ValueNotifier<int> notifier;
+
+  @override
+  State<PositiveIntegerSetting> createState() => _PositiveIntegerSettingState();
+}
+
+class _PositiveIntegerSettingState extends State<PositiveIntegerSetting> {
+  late final TextEditingController _textEditingController;
+
+  @override
+  void initState() {
+    super.initState();
+    _textEditingController = TextEditingController()
+      ..text = widget.notifier.value.toString();
```

Comment:
does this text controller need to be updated when the value of widget.notifier changes? for example if widget.notifier changed outside the context of this widget?